signal-hook-registry/src/lib.rs

```diff
@@ -167,6 +213,38 @@ impl GlobalData {
     }
 }
 
+#[cfg(windows)]
+extern "C" fn handler(sig: c_int) {
+    if sig != SIGFPE {
+        // Windows CRT `signal` resets handler every time, unless for SIGFPE.
+        // Reregister the handler to retain maximal compatibility.
+        // Problems:
+        // - It's racy. But this is inevitably racy in Windows.
+        // - Interacts poorly with handlers outside signal-hook-registry.
+        unsafe {
+            libc::signal(sig, handler as sighandler_t);
```

Comment:
I guess there should be some error handling here.

I'm not 100% sure what to do if there's an error, though. Maybe doing `libc::abort()` and commenting that there's nothing better to be done in such case and that it should probably not happen anyway (because we already *had* the signal), but I'd prefer not to leave the application in somewhat half-broken state.